examples/src/main/scala/io/finch/oauth2/Main.scala

```diff
@@ -0,0 +1,54 @@
+package io.finch.oauth2
+
+import com.twitter.finagle._
+import com.twitter.finagle.oauth2.{AuthInfo, GrantHandlerResult}
+import com.twitter.util.Await
+import io.finch._
+import io.finch.circe._
+import io.circe.generic.auto._
+
+/**
+  * A simple example of finch-oauth2 usage
+  *
+  * Use the following sbt command to run the application.
+  *
+  * {{{
+  *   $ sbt 'examples/runMain io.finch.oauth2.Main'
+  * }}}
+  *
+  * Use the following HTTPie commands to test endpoints.
+  *
+  * {{{
+  *   $ http POST :8081/oauth2/tokens Authorization:'OAuth dXNlcl9pZDp1c2VyX3NlY3JldA=='\
+  *     grant_type==client_credentials
+  *
+  *   $ http GET :8081/oauth2/tokens grant_type==password username==user_name\
+  *     password==user_password client_id==user_id
+  *
+  *   $ http GET :8081/oauth2/tokens grant_type==authorization_code code==user_auth_code client_id==user_id
+  *
+  *   $ http GET :8081/oauth2/users access_token=='AT-5b0e7e3b-943f-479f-beab-7814814d0315'
+  *
+  *   $ http GET :8081/oauth2/tokens client_id==user_id grant_type==refresh_token\
+  *     refresh_token=='RT-7e1bbf43-e7ba-4a8a-a38e-baf62ce3ceae'
+  *
+  *   $ http GET :8081/oauth2/unprotected
+  * }}}
+  */
+object Main extends App {
+
+  val users: Endpoint[OAuthUser] = get("oauth2" :: "users" :: authorize(InMemoryDataHandler)) {
+    ai: AuthInfo[OAuthUser] => Ok(ai.user)
+  }
+
+  val tokens = get("oauth2" :: "tokens" :: issueAccessToken(InMemoryDataHandler)) {
```

Comment:
I'd name this `val auth: Endpoint[GrantHandlerResult] = post("users" :: "auth" :: ...)`. Note `post` instead of `get`.

Also you don't need to actually map this endpoint: `x => Ok(x)` is usually redundant.